ocaml/typing/typecore.ml

```diff
@@ -1094,6 +1115,11 @@ let maybe_add_pattern_variables_ghost loc_let env pv =
        end
     ) pv env
 
+let add_fns_defined_by_letrec env pvs =
+  List.fold_right
```

Comment:
Nit: I don't think we care about the difference in semantics between `fold_right` and `fold_left`, so we can probably `fold_left` instead. (When I see a `fold_right`, I generally assume there's some significance to it that it's important I not miss.)